File: frontend/src/types/task.ts
export interface Task {
  id: string
  name: string
  description: string
  course_id: string
  teacher_id: string
  start_date: string
  end_date: string
  status: string
  created_at: string
  updated_at: string
  teacher?: {
    id: string
    name: string
    email: string
  }
  student_count?: number
  completed_count?: number
}

export interface StudentTask {
  id: string
  task_id: string
  student_id: string
  status: string
  progress: number
  created_at: string
  updated_at: string
  task?: Task
  student?: {
    id: string
    name: string
    student_id: string
  }
}

export interface CreateTaskRequest {
  name: string
  description: string
  course_id: string
  start_date: string
  end_date: string
}

export interface AssignTaskRequest {
  student_ids: string[]
}

export interface TaskStatistics {
  total_students: number
  completed: number
  processing: number
  pending: number
}

File: internal/models/task.go
package models

import (
	"time"
)

type Task struct {
	ID          string    `json:"id" gorm:"primaryKey"`
	Name        string    `json:"name" gorm:"not null"`
	Description string    `json:"description"`
	CourseID    string    `json:"course_id" gorm:"not null"`
	TeacherID   string    `json:"teacher_id" gorm:"not null"`
	StartDate   time.Time `json:"start_date" gorm:"not null"`
	EndDate     time.Time `json:"end_date" gorm:"not null"`
	Status      string    `json:"status" gorm:"not null"` // active, completed, archived
	CreatedAt   time.Time `json:"created_at" gorm:"autoCreateTime"`
	UpdatedAt   time.Time `json:"updated_at" gorm:"autoUpdateTime"`
	
	// 移除关联关系，避免迁移问题
	// Teacher     Teacher    `json:"teacher" gorm:"foreignKey:TeacherID"`
	// StudentTasks []StudentTask `json:"student_tasks" gorm:"foreignKey:TaskID"`
}

type StudentTask struct {
	ID        string    `json:"id" gorm:"primaryKey"`
	TaskID    string    `json:"task_id" gorm:"not null"`
	StudentID string    `json:"student_id" gorm:"not null"`
	Status    string    `json:"status" gorm:"not null"` // pending, processing, completed
	Progress  int       `json:"progress" gorm:"not null;default:0"`
	CreatedAt time.Time `json:"created_at" gorm:"autoCreateTime"`
	UpdatedAt time.Time `json:"updated_at" gorm:"autoUpdateTime"`
	
	// 移除关联关系，避免迁移问题
	// Task      Task      `json:"task" gorm:"foreignKey:TaskID"`
	// Student   Student   `json:"student" gorm:"foreignKey:StudentID"`
	// Evidences []Evidence `json:"evidences" gorm:"foreignKey:StudentTaskID"`
}